app.rb

```diff
@@ -0,0 +1,97 @@
+require_relative './student'
+require_relative './teacher'
+require_relative './book'
+require_relative './classroom'
+require_relative './rental'
+
+class App
+  attr_accessor :books, :students, :teachers, :rentals
+
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def display_all_books
+    @books.each_with_index do |book, index|
+      puts "#{index}) Title: '#{book.title}', Author: '#{book.author}'"
+    end
+  end
+
+  def display_all_people
+    @people.each_with_index do |people, index|
+      puts "#{index} Name: #{people.name}, ID: #{people.id}, Age: #{people.age}"
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student(1) or a teacher(2)? [Input the number]: '
+    person_option = gets.chomp.to_i
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    if person_option == 1
+      print 'Classroom: '
+      classroom_ = gets.chomp
+      print 'Has parent permmission? [Y/N]: '
+      parent_permmission = gets.chomp.downcase == 'y'
+      create_student(name, age, parent_permmission, classroom_)
+    else
+      print 'Specialization: '
+      specialization = gets.chomp
+      create_teacher(name, age, specialization)
+    end
+    puts 'Person created successfully'
+  end
+
+  def create_student(name, age, parent_permmission, classroom)
+    classroom = Classroom.new(classroom)
+    student = Student.new(classroom, age, name, parent_permmission)
+    @people << student
+  end
+
+  def create_teacher(name, age, specialization)
+    teacher = Teacher.new(specialization, age, name)
+    @people << teacher
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    book = Book.new(title, author)
+    @books << book
+    puts 'Book created successfully'
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number'
+    display_all_books
+    selected_book_option = gets.chomp.to_i
+    display_all_people
+    selected_person_option = gets.chomp.to_i
+    print 'Date: '
+    date = gets.chomp
+    book = @books[selected_book_option]
+    person = @people[selected_person_option]
+    rental = Rental.new(date, book, person)
```

Comment:
- [ ] I am unable to create a rental because in your Rental class you defined `rental` instead of `rentals` for book and person. In your Rental class kindly fix that.

  ```
  def initialize(date, person, book)
    ...
    person.rentals << self

    ...
    book.rentals << self
  end
  ```

  ![image](https://user-images.githubusercontent.com/36542986/200128758-657528f0-b156-48df-8a73-c8ab69212c39.png)

- [ ] Nothing happens when I try to list the rentals for a person, because of the order of the arguments here. You should follow the order you defined.

  ```
  rental = Rental.new(date, person, book)
  ```

  ![wws](https://user-images.githubusercontent.com/36542986/200128776-87eca8b6-f6a0-45ca-9043-4a579873c7dc.jpg)

Reply:
Thanks, @TedLivist  for all your reviews and kind recommendations and fix. I am really delighted with this honest review. 

I have fixed all the issues involved. Cheers 🚀 😄